{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "50650c12",
   "metadata": {},
   "source": [
    "# Cell Line Metadata\n",
    "\n",
    "Table of the cell line names with the DepMap_ID, and the \"cleaned\" names (name all in capital and with no special characters).\n",
    "\n",
    "  *  Read in-house RNA-seq dataset experiment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "8790b95c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import collections\n",
    "\n",
    "import cmapPy\n",
    "import cmapPy.pandasGEXpress.parse as parse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c8f4337e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\fuzan\\Documents\\code\\cell_line_authen\\Cell Line Authentification\\Final pipeline\n"
     ]
    }
   ],
   "source": [
    "print(os.getcwd())\n",
    "# os.chdir(\"C:/Users/fuzan/Documents/code/cell_line_authen/Cell Line Authentification/Final pipeline/\")\n",
    "# print(os.getcwd())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f40db565",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Files\n",
    "\n",
    "# In-house RNA-seq dataset\n",
    "query_expr_file = \"./input_data/2022-02-10_heatmap_log2_TPM_r113153x1115.gctx\"\n",
    "\n",
    "# Read the sample_info containing the DepMap_ID and the cell line name\n",
    "sample_info_file = \"./input_data/sample_info.csv\" \n",
    "\n",
    "# Read the metadata from Lab Guru to comapr sample and real cell line name\n",
    "metadata_file = \"./input_data/H202SC21053155 2021-05-25 next gen sequencing experiment annotations template - DS 5-25-2021_36 samples.csv\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fb26c1d9",
   "metadata": {},
   "source": [
    "## Read in-house RNA-seq dataset experiment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3dfc5c5e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(113153, 1115)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>cid</th>\n",
       "      <th>H101SC20021233_2_ASKA_1</th>\n",
       "      <th>H101SC20021233_2_ASKA_10</th>\n",
       "      <th>H101SC20021233_2_ASKA_11</th>\n",
       "      <th>H101SC20021233_2_ASKA_12</th>\n",
       "      <th>H101SC20021233_2_ASKA_2</th>\n",
       "      <th>H101SC20021233_2_ASKA_3</th>\n",
       "      <th>H101SC20021233_2_ASKA_4</th>\n",
       "      <th>H101SC20021233_2_ASKA_5</th>\n",
       "      <th>H101SC20021233_2_ASKA_6</th>\n",
       "      <th>H101SC20021233_2_ASKA_7</th>\n",
       "      <th>...</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G7_38</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G7_40</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G7_44</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G7_51</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G8_18</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G8_27</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G8_29</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G8_56</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G8_7</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G8_9</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <td>3.573865</td>\n",
       "      <td>3.354822</td>\n",
       "      <td>3.463820</td>\n",
       "      <td>3.409673</td>\n",
       "      <td>3.644708</td>\n",
       "      <td>3.641184</td>\n",
       "      <td>3.584009</td>\n",
       "      <td>3.544834</td>\n",
       "      <td>3.614835</td>\n",
       "      <td>3.391639</td>\n",
       "      <td>...</td>\n",
       "      <td>2.226621</td>\n",
       "      <td>3.079372</td>\n",
       "      <td>3.073363</td>\n",
       "      <td>2.472898</td>\n",
       "      <td>2.017262</td>\n",
       "      <td>1.918629</td>\n",
       "      <td>2.996959</td>\n",
       "      <td>3.037888</td>\n",
       "      <td>1.941835</td>\n",
       "      <td>3.264794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000005</th>\n",
       "      <td>-3.470722</td>\n",
       "      <td>-3.938031</td>\n",
       "      <td>-3.960276</td>\n",
       "      <td>-3.923771</td>\n",
       "      <td>-4.029152</td>\n",
       "      <td>-4.060926</td>\n",
       "      <td>-3.977351</td>\n",
       "      <td>-3.997794</td>\n",
       "      <td>-3.992024</td>\n",
       "      <td>-3.971337</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.181395</td>\n",
       "      <td>-2.822084</td>\n",
       "      <td>-2.659540</td>\n",
       "      <td>-2.784278</td>\n",
       "      <td>-2.633618</td>\n",
       "      <td>-2.571648</td>\n",
       "      <td>-2.848423</td>\n",
       "      <td>-2.607749</td>\n",
       "      <td>-2.545574</td>\n",
       "      <td>-2.682902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000419</th>\n",
       "      <td>6.481495</td>\n",
       "      <td>6.424484</td>\n",
       "      <td>6.413052</td>\n",
       "      <td>6.470856</td>\n",
       "      <td>6.425018</td>\n",
       "      <td>6.438981</td>\n",
       "      <td>6.470403</td>\n",
       "      <td>6.351199</td>\n",
       "      <td>6.459817</td>\n",
       "      <td>6.557379</td>\n",
       "      <td>...</td>\n",
       "      <td>6.051025</td>\n",
       "      <td>5.993428</td>\n",
       "      <td>6.074045</td>\n",
       "      <td>5.800958</td>\n",
       "      <td>5.759847</td>\n",
       "      <td>5.817563</td>\n",
       "      <td>6.176519</td>\n",
       "      <td>6.081125</td>\n",
       "      <td>5.945285</td>\n",
       "      <td>5.865779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000457</th>\n",
       "      <td>1.574731</td>\n",
       "      <td>1.762096</td>\n",
       "      <td>1.659323</td>\n",
       "      <td>1.607489</td>\n",
       "      <td>1.319769</td>\n",
       "      <td>1.298866</td>\n",
       "      <td>1.528153</td>\n",
       "      <td>1.322604</td>\n",
       "      <td>1.440083</td>\n",
       "      <td>1.560199</td>\n",
       "      <td>...</td>\n",
       "      <td>2.447656</td>\n",
       "      <td>2.408158</td>\n",
       "      <td>2.475372</td>\n",
       "      <td>2.931849</td>\n",
       "      <td>3.037073</td>\n",
       "      <td>3.194029</td>\n",
       "      <td>2.400180</td>\n",
       "      <td>2.761051</td>\n",
       "      <td>3.333862</td>\n",
       "      <td>2.629969</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000460</th>\n",
       "      <td>1.742566</td>\n",
       "      <td>1.869973</td>\n",
       "      <td>1.839150</td>\n",
       "      <td>1.854894</td>\n",
       "      <td>1.671424</td>\n",
       "      <td>1.664413</td>\n",
       "      <td>1.752028</td>\n",
       "      <td>1.614235</td>\n",
       "      <td>1.690203</td>\n",
       "      <td>1.704259</td>\n",
       "      <td>...</td>\n",
       "      <td>2.249911</td>\n",
       "      <td>2.444569</td>\n",
       "      <td>2.470564</td>\n",
       "      <td>2.858309</td>\n",
       "      <td>2.664449</td>\n",
       "      <td>2.899413</td>\n",
       "      <td>2.181866</td>\n",
       "      <td>2.492191</td>\n",
       "      <td>2.873590</td>\n",
       "      <td>2.520076</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1115 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "cid              H101SC20021233_2_ASKA_1  H101SC20021233_2_ASKA_10  \\\n",
       "rid                                                                  \n",
       "ENSG00000000003                 3.573865                  3.354822   \n",
       "ENSG00000000005                -3.470722                 -3.938031   \n",
       "ENSG00000000419                 6.481495                  6.424484   \n",
       "ENSG00000000457                 1.574731                  1.762096   \n",
       "ENSG00000000460                 1.742566                  1.869973   \n",
       "\n",
       "cid              H101SC20021233_2_ASKA_11  H101SC20021233_2_ASKA_12  \\\n",
       "rid                                                                   \n",
       "ENSG00000000003                  3.463820                  3.409673   \n",
       "ENSG00000000005                 -3.960276                 -3.923771   \n",
       "ENSG00000000419                  6.413052                  6.470856   \n",
       "ENSG00000000457                  1.659323                  1.607489   \n",
       "ENSG00000000460                  1.839150                  1.854894   \n",
       "\n",
       "cid              H101SC20021233_2_ASKA_2  H101SC20021233_2_ASKA_3  \\\n",
       "rid                                                                 \n",
       "ENSG00000000003                 3.644708                 3.641184   \n",
       "ENSG00000000005                -4.029152                -4.060926   \n",
       "ENSG00000000419                 6.425018                 6.438981   \n",
       "ENSG00000000457                 1.319769                 1.298866   \n",
       "ENSG00000000460                 1.671424                 1.664413   \n",
       "\n",
       "cid              H101SC20021233_2_ASKA_4  H101SC20021233_2_ASKA_5  \\\n",
       "rid                                                                 \n",
       "ENSG00000000003                 3.584009                 3.544834   \n",
       "ENSG00000000005                -3.977351                -3.997794   \n",
       "ENSG00000000419                 6.470403                 6.351199   \n",
       "ENSG00000000457                 1.528153                 1.322604   \n",
       "ENSG00000000460                 1.752028                 1.614235   \n",
       "\n",
       "cid              H101SC20021233_2_ASKA_6  H101SC20021233_2_ASKA_7  ...  \\\n",
       "rid                                                                ...   \n",
       "ENSG00000000003                 3.614835                 3.391639  ...   \n",
       "ENSG00000000005                -3.992024                -3.971337  ...   \n",
       "ENSG00000000419                 6.459817                 6.557379  ...   \n",
       "ENSG00000000457                 1.440083                 1.560199  ...   \n",
       "ENSG00000000460                 1.690203                 1.704259  ...   \n",
       "\n",
       "cid              PH-FTX-IVP-2021-006-007_G7_38  PH-FTX-IVP-2021-006-007_G7_40  \\\n",
       "rid                                                                             \n",
       "ENSG00000000003                       2.226621                       3.079372   \n",
       "ENSG00000000005                      -3.181395                      -2.822084   \n",
       "ENSG00000000419                       6.051025                       5.993428   \n",
       "ENSG00000000457                       2.447656                       2.408158   \n",
       "ENSG00000000460                       2.249911                       2.444569   \n",
       "\n",
       "cid              PH-FTX-IVP-2021-006-007_G7_44  PH-FTX-IVP-2021-006-007_G7_51  \\\n",
       "rid                                                                             \n",
       "ENSG00000000003                       3.073363                       2.472898   \n",
       "ENSG00000000005                      -2.659540                      -2.784278   \n",
       "ENSG00000000419                       6.074045                       5.800958   \n",
       "ENSG00000000457                       2.475372                       2.931849   \n",
       "ENSG00000000460                       2.470564                       2.858309   \n",
       "\n",
       "cid              PH-FTX-IVP-2021-006-007_G8_18  PH-FTX-IVP-2021-006-007_G8_27  \\\n",
       "rid                                                                             \n",
       "ENSG00000000003                       2.017262                       1.918629   \n",
       "ENSG00000000005                      -2.633618                      -2.571648   \n",
       "ENSG00000000419                       5.759847                       5.817563   \n",
       "ENSG00000000457                       3.037073                       3.194029   \n",
       "ENSG00000000460                       2.664449                       2.899413   \n",
       "\n",
       "cid              PH-FTX-IVP-2021-006-007_G8_29  PH-FTX-IVP-2021-006-007_G8_56  \\\n",
       "rid                                                                             \n",
       "ENSG00000000003                       2.996959                       3.037888   \n",
       "ENSG00000000005                      -2.848423                      -2.607749   \n",
       "ENSG00000000419                       6.176519                       6.081125   \n",
       "ENSG00000000457                       2.400180                       2.761051   \n",
       "ENSG00000000460                       2.181866                       2.492191   \n",
       "\n",
       "cid              PH-FTX-IVP-2021-006-007_G8_7  PH-FTX-IVP-2021-006-007_G8_9  \n",
       "rid                                                                          \n",
       "ENSG00000000003                      1.941835                      3.264794  \n",
       "ENSG00000000005                     -2.545574                     -2.682902  \n",
       "ENSG00000000419                      5.945285                      5.865779  \n",
       "ENSG00000000457                      3.333862                      2.629969  \n",
       "ENSG00000000460                      2.873590                      2.520076  \n",
       "\n",
       "[5 rows x 1115 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read in-house RNA seq dataset \n",
    "\n",
    "query_expr_gctx = parse.parse(query_expr_file)\n",
    "query_expr = query_expr_gctx.data_df\n",
    "\n",
    "print(query_expr.shape)\n",
    "query_expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0aca9842",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(57771, 1115)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>cid</th>\n",
       "      <th>H101SC20021233_2_ASKA_1</th>\n",
       "      <th>H101SC20021233_2_ASKA_10</th>\n",
       "      <th>H101SC20021233_2_ASKA_11</th>\n",
       "      <th>H101SC20021233_2_ASKA_12</th>\n",
       "      <th>H101SC20021233_2_ASKA_2</th>\n",
       "      <th>H101SC20021233_2_ASKA_3</th>\n",
       "      <th>H101SC20021233_2_ASKA_4</th>\n",
       "      <th>H101SC20021233_2_ASKA_5</th>\n",
       "      <th>H101SC20021233_2_ASKA_6</th>\n",
       "      <th>H101SC20021233_2_ASKA_7</th>\n",
       "      <th>...</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G7_38</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G7_40</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G7_44</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G7_51</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G8_18</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G8_27</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G8_29</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G8_56</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G8_7</th>\n",
       "      <th>PH-FTX-IVP-2021-006-007_G8_9</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <td>3.573865</td>\n",
       "      <td>3.354822</td>\n",
       "      <td>3.463820</td>\n",
       "      <td>3.409673</td>\n",
       "      <td>3.644708</td>\n",
       "      <td>3.641184</td>\n",
       "      <td>3.584009</td>\n",
       "      <td>3.544834</td>\n",
       "      <td>3.614835</td>\n",
       "      <td>3.391639</td>\n",
       "      <td>...</td>\n",
       "      <td>2.226621</td>\n",
       "      <td>3.079372</td>\n",
       "      <td>3.073363</td>\n",
       "      <td>2.472898</td>\n",
       "      <td>2.017262</td>\n",
       "      <td>1.918629</td>\n",
       "      <td>2.996959</td>\n",
       "      <td>3.037888</td>\n",
       "      <td>1.941835</td>\n",
       "      <td>3.264794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000005</th>\n",
       "      <td>-3.470722</td>\n",
       "      <td>-3.938031</td>\n",
       "      <td>-3.960276</td>\n",
       "      <td>-3.923771</td>\n",
       "      <td>-4.029152</td>\n",
       "      <td>-4.060926</td>\n",
       "      <td>-3.977351</td>\n",
       "      <td>-3.997794</td>\n",
       "      <td>-3.992024</td>\n",
       "      <td>-3.971337</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.181395</td>\n",
       "      <td>-2.822084</td>\n",
       "      <td>-2.659540</td>\n",
       "      <td>-2.784278</td>\n",
       "      <td>-2.633618</td>\n",
       "      <td>-2.571648</td>\n",
       "      <td>-2.848423</td>\n",
       "      <td>-2.607749</td>\n",
       "      <td>-2.545574</td>\n",
       "      <td>-2.682902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000419</th>\n",
       "      <td>6.481495</td>\n",
       "      <td>6.424484</td>\n",
       "      <td>6.413052</td>\n",
       "      <td>6.470856</td>\n",
       "      <td>6.425018</td>\n",
       "      <td>6.438981</td>\n",
       "      <td>6.470403</td>\n",
       "      <td>6.351199</td>\n",
       "      <td>6.459817</td>\n",
       "      <td>6.557379</td>\n",
       "      <td>...</td>\n",
       "      <td>6.051025</td>\n",
       "      <td>5.993428</td>\n",
       "      <td>6.074045</td>\n",
       "      <td>5.800958</td>\n",
       "      <td>5.759847</td>\n",
       "      <td>5.817563</td>\n",
       "      <td>6.176519</td>\n",
       "      <td>6.081125</td>\n",
       "      <td>5.945285</td>\n",
       "      <td>5.865779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000457</th>\n",
       "      <td>1.574731</td>\n",
       "      <td>1.762096</td>\n",
       "      <td>1.659323</td>\n",
       "      <td>1.607489</td>\n",
       "      <td>1.319769</td>\n",
       "      <td>1.298866</td>\n",
       "      <td>1.528153</td>\n",
       "      <td>1.322604</td>\n",
       "      <td>1.440083</td>\n",
       "      <td>1.560199</td>\n",
       "      <td>...</td>\n",
       "      <td>2.447656</td>\n",
       "      <td>2.408158</td>\n",
       "      <td>2.475372</td>\n",
       "      <td>2.931849</td>\n",
       "      <td>3.037073</td>\n",
       "      <td>3.194029</td>\n",
       "      <td>2.400180</td>\n",
       "      <td>2.761051</td>\n",
       "      <td>3.333862</td>\n",
       "      <td>2.629969</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000460</th>\n",
       "      <td>1.742566</td>\n",
       "      <td>1.869973</td>\n",
       "      <td>1.839150</td>\n",
       "      <td>1.854894</td>\n",
       "      <td>1.671424</td>\n",
       "      <td>1.664413</td>\n",
       "      <td>1.752028</td>\n",
       "      <td>1.614235</td>\n",
       "      <td>1.690203</td>\n",
       "      <td>1.704259</td>\n",
       "      <td>...</td>\n",
       "      <td>2.249911</td>\n",
       "      <td>2.444569</td>\n",
       "      <td>2.470564</td>\n",
       "      <td>2.858309</td>\n",
       "      <td>2.664449</td>\n",
       "      <td>2.899413</td>\n",
       "      <td>2.181866</td>\n",
       "      <td>2.492191</td>\n",
       "      <td>2.873590</td>\n",
       "      <td>2.520076</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1115 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "cid              H101SC20021233_2_ASKA_1  H101SC20021233_2_ASKA_10  \\\n",
       "rid                                                                  \n",
       "ENSG00000000003                 3.573865                  3.354822   \n",
       "ENSG00000000005                -3.470722                 -3.938031   \n",
       "ENSG00000000419                 6.481495                  6.424484   \n",
       "ENSG00000000457                 1.574731                  1.762096   \n",
       "ENSG00000000460                 1.742566                  1.869973   \n",
       "\n",
       "cid              H101SC20021233_2_ASKA_11  H101SC20021233_2_ASKA_12  \\\n",
       "rid                                                                   \n",
       "ENSG00000000003                  3.463820                  3.409673   \n",
       "ENSG00000000005                 -3.960276                 -3.923771   \n",
       "ENSG00000000419                  6.413052                  6.470856   \n",
       "ENSG00000000457                  1.659323                  1.607489   \n",
       "ENSG00000000460                  1.839150                  1.854894   \n",
       "\n",
       "cid              H101SC20021233_2_ASKA_2  H101SC20021233_2_ASKA_3  \\\n",
       "rid                                                                 \n",
       "ENSG00000000003                 3.644708                 3.641184   \n",
       "ENSG00000000005                -4.029152                -4.060926   \n",
       "ENSG00000000419                 6.425018                 6.438981   \n",
       "ENSG00000000457                 1.319769                 1.298866   \n",
       "ENSG00000000460                 1.671424                 1.664413   \n",
       "\n",
       "cid              H101SC20021233_2_ASKA_4  H101SC20021233_2_ASKA_5  \\\n",
       "rid                                                                 \n",
       "ENSG00000000003                 3.584009                 3.544834   \n",
       "ENSG00000000005                -3.977351                -3.997794   \n",
       "ENSG00000000419                 6.470403                 6.351199   \n",
       "ENSG00000000457                 1.528153                 1.322604   \n",
       "ENSG00000000460                 1.752028                 1.614235   \n",
       "\n",
       "cid              H101SC20021233_2_ASKA_6  H101SC20021233_2_ASKA_7  ...  \\\n",
       "rid                                                                ...   \n",
       "ENSG00000000003                 3.614835                 3.391639  ...   \n",
       "ENSG00000000005                -3.992024                -3.971337  ...   \n",
       "ENSG00000000419                 6.459817                 6.557379  ...   \n",
       "ENSG00000000457                 1.440083                 1.560199  ...   \n",
       "ENSG00000000460                 1.690203                 1.704259  ...   \n",
       "\n",
       "cid              PH-FTX-IVP-2021-006-007_G7_38  PH-FTX-IVP-2021-006-007_G7_40  \\\n",
       "rid                                                                             \n",
       "ENSG00000000003                       2.226621                       3.079372   \n",
       "ENSG00000000005                      -3.181395                      -2.822084   \n",
       "ENSG00000000419                       6.051025                       5.993428   \n",
       "ENSG00000000457                       2.447656                       2.408158   \n",
       "ENSG00000000460                       2.249911                       2.444569   \n",
       "\n",
       "cid              PH-FTX-IVP-2021-006-007_G7_44  PH-FTX-IVP-2021-006-007_G7_51  \\\n",
       "rid                                                                             \n",
       "ENSG00000000003                       3.073363                       2.472898   \n",
       "ENSG00000000005                      -2.659540                      -2.784278   \n",
       "ENSG00000000419                       6.074045                       5.800958   \n",
       "ENSG00000000457                       2.475372                       2.931849   \n",
       "ENSG00000000460                       2.470564                       2.858309   \n",
       "\n",
       "cid              PH-FTX-IVP-2021-006-007_G8_18  PH-FTX-IVP-2021-006-007_G8_27  \\\n",
       "rid                                                                             \n",
       "ENSG00000000003                       2.017262                       1.918629   \n",
       "ENSG00000000005                      -2.633618                      -2.571648   \n",
       "ENSG00000000419                       5.759847                       5.817563   \n",
       "ENSG00000000457                       3.037073                       3.194029   \n",
       "ENSG00000000460                       2.664449                       2.899413   \n",
       "\n",
       "cid              PH-FTX-IVP-2021-006-007_G8_29  PH-FTX-IVP-2021-006-007_G8_56  \\\n",
       "rid                                                                             \n",
       "ENSG00000000003                       2.996959                       3.037888   \n",
       "ENSG00000000005                      -2.848423                      -2.607749   \n",
       "ENSG00000000419                       6.176519                       6.081125   \n",
       "ENSG00000000457                       2.400180                       2.761051   \n",
       "ENSG00000000460                       2.181866                       2.492191   \n",
       "\n",
       "cid              PH-FTX-IVP-2021-006-007_G8_7  PH-FTX-IVP-2021-006-007_G8_9  \n",
       "rid                                                                          \n",
       "ENSG00000000003                      1.941835                      3.264794  \n",
       "ENSG00000000005                     -2.545574                     -2.682902  \n",
       "ENSG00000000419                      5.945285                      5.865779  \n",
       "ENSG00000000457                      3.333862                      2.629969  \n",
       "ENSG00000000460                      2.873590                      2.520076  \n",
       "\n",
       "[5 rows x 1115 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remove ENSMUSG (mouse) genes\n",
    "query_expr = query_expr[query_expr.index.str.contains(\"ENSMUSG\")==False]\n",
    "\n",
    "print(query_expr.shape)\n",
    "query_expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "b98e6fa0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NAs in the indexes:  Counter({False: 57771})\n",
      "NAs in the columns:  Counter({False: 1115})\n",
      "NAs in DataFrame:  693252\n",
      "There is 12 columns with only NAs in this Dataframe.\n",
      "Names of columns/samples with all NAs in DataFrame: \n",
      " ['H202SC20082867_DMSO_1', 'H202SC20082867_DMSO_2', 'H202SC20082867_F034_11', 'H202SC20082867_F034_12', 'H202SC20082867_F034_21', 'H202SC20082867_F034_22', 'H202SC20082867_F2256_11', 'H202SC20082867_F2256_12', 'H202SC20082867_F2256_21', 'H202SC20082867_F2256_22', 'H202SC20082867_F4472_21', 'H202SC20082867_F4472_22']\n",
      "New # of NAs in DataFrame:  0\n"
     ]
    }
   ],
   "source": [
    "# Check for NAs in index\n",
    "print('NAs in the indexes: ', collections.Counter(list(query_expr.index.isna())))\n",
    "\n",
    "# Check for NAs in columns\n",
    "print('NAs in the columns: ', collections.Counter(list(query_expr.columns.isna())))\n",
    "\n",
    "# Check for NaN in DataFrame\n",
    "print('NAs in DataFrame: ', query_expr.isnull().sum().sum())\n",
    "\n",
    "# View columns with all NAs (Mouse samples) if any\n",
    "NA_col_query_expr = collections.Counter(query_expr.isnull().all())\n",
    "if NA_col_query_expr[True] ==0:\n",
    "    print(\"There is no columns with only NAs in the Dataframe.\")\n",
    "else:\n",
    "    print('There is', NA_col_query_expr[True] , 'columns with only NAs in this Dataframe.')\n",
    "    NA_samples = list(query_expr[query_expr.columns[query_expr.isnull().all()]].columns)\n",
    "    print('Names of columns/samples with all NAs in DataFrame: \\n', NA_samples)\n",
    "    # Drop columns with all NAs in query_expr\n",
    "    query_expr = query_expr.dropna(axis='columns')\n",
    "\n",
    "# Sanity check\n",
    "print('New # of NAs in DataFrame: ', query_expr.isnull().sum().sum())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d286ca1e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save file to csv\n",
    "query_expr.to_csv('input_data/query_expr.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "84bcf6e5",
   "metadata": {},
   "source": [
    "## Add cell line name to the sample name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "7cca8997",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1103, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>bio_context_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>H101SC20021233_2_ASKA_1</td>\n",
       "      <td>ASKA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>H101SC20021233_2_ASKA_10</td>\n",
       "      <td>ASKA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>H101SC20021233_2_ASKA_11</td>\n",
       "      <td>ASKA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>H101SC20021233_2_ASKA_12</td>\n",
       "      <td>ASKA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>H101SC20021233_2_ASKA_2</td>\n",
       "      <td>ASKA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     sample bio_context_id\n",
       "0   H101SC20021233_2_ASKA_1           ASKA\n",
       "1  H101SC20021233_2_ASKA_10           ASKA\n",
       "2  H101SC20021233_2_ASKA_11           ASKA\n",
       "3  H101SC20021233_2_ASKA_12           ASKA\n",
       "4   H101SC20021233_2_ASKA_2           ASKA"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read the metadata from 10_heatmap_log2_TPM_r113153x1115.gctx to compare the cell line sample \n",
    "# and the cell line with the highest correlation\n",
    "\n",
    "metadata = pd.DataFrame(query_expr_gctx.col_metadata_df['bio_context_id'])\n",
    "metadata.index.name = \"sample\"\n",
    "metadata.drop(NA_samples, inplace= True) # Drop the samples with all NAs samples (Mouse samples)\n",
    "metadata.reset_index(inplace=True)\n",
    "\n",
    "print(metadata.shape)\n",
    "metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "0253212b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NAs in the indexes:  Counter({False: 1103})\n",
      "NAs in the columns:  Counter({False: 2})\n",
      "NAs in DataFrame:  34\n",
      "Samples with no annotation: \n",
      " [225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 255, 256, 257, 258]\n",
      "New # of NAs in DataFrame:  0\n",
      "Dim of the DataFrame:  (1103, 2)\n"
     ]
    }
   ],
   "source": [
    "# Check for NAs in index\n",
    "print('NAs in the indexes: ', collections.Counter(list(metadata.index.isna())))\n",
    "\n",
    "# Check for NAs in columns\n",
    "print('NAs in the columns: ', collections.Counter(list(metadata.columns.isna())))\n",
    "\n",
    "# Check for NaN in DataFrame\n",
    "print('NAs in DataFrame: ', metadata.isnull().sum().sum())\n",
    "\n",
    "# View samples with no annotation if any\n",
    "sample_wth_no_annot = [sample for sample, row in metadata.iterrows() if row.isnull().any()]\n",
    "if sample_wth_no_annot==[]:\n",
    "    print('All samples have an annotation.')\n",
    "else:\n",
    "    print('Samples with no annotation: \\n', sample_wth_no_annot)\n",
    "    # Replace samples with no annotation by 'NoAnnotation' in metadata if NAs\n",
    "    metadata.fillna('NOANNOTATION', inplace=True)\n",
    "\n",
    "# Sanity Check\n",
    "print('New # of NAs in DataFrame: ', metadata.isnull().sum().sum())\n",
    "print('Dim of the DataFrame: ', metadata.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e5aadb12",
   "metadata": {},
   "source": [
    "## Add \"cleaned\" cell line name "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ba3c3b42",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1103, 3)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\fuzan\\Miniconda3\\envs\\compbio\\lib\\site-packages\\ipykernel_launcher.py:5: FutureWarning: The default value of regex will change from True to False in a future version.\n",
      "  \"\"\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>bio_context_id</th>\n",
       "      <th>cleaned_bio_context_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>H101SC20021233_2_ASKA_1</td>\n",
       "      <td>ASKA</td>\n",
       "      <td>ASKA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>H101SC20021233_2_ASKA_10</td>\n",
       "      <td>ASKA</td>\n",
       "      <td>ASKA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>H101SC20021233_2_ASKA_11</td>\n",
       "      <td>ASKA</td>\n",
       "      <td>ASKA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>H101SC20021233_2_ASKA_12</td>\n",
       "      <td>ASKA</td>\n",
       "      <td>ASKA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>H101SC20021233_2_ASKA_2</td>\n",
       "      <td>ASKA</td>\n",
       "      <td>ASKA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     sample bio_context_id cleaned_bio_context_id\n",
       "0   H101SC20021233_2_ASKA_1           ASKA                   ASKA\n",
       "1  H101SC20021233_2_ASKA_10           ASKA                   ASKA\n",
       "2  H101SC20021233_2_ASKA_11           ASKA                   ASKA\n",
       "3  H101SC20021233_2_ASKA_12           ASKA                   ASKA\n",
       "4   H101SC20021233_2_ASKA_2           ASKA                   ASKA"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata[\"cleaned_bio_context_id\"] = metadata[\"bio_context_id\"] \n",
    "# 1) force all names to be uppercase \n",
    "metadata[\"cleaned_bio_context_id\"] = metadata[\"cleaned_bio_context_id\"].str.upper()\n",
    "# 2) remove all special characters before comparing\n",
    "metadata[\"cleaned_bio_context_id\"] = metadata[\"cleaned_bio_context_id\"].str.replace('[^A-Za-z0-9]+','')\n",
    "# 3) remove LabGuru_cleaned_annot_cl\n",
    "metadata[\"cleaned_bio_context_id\"] = metadata[\"cleaned_bio_context_id\"].str.replace('NCI','')\n",
    "\n",
    "print(metadata.shape)\n",
    "metadata.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "26622221",
   "metadata": {},
   "source": [
    "## Add DepMap_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "fe358b9a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1825, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DepMap_ID</th>\n",
       "      <th>stripped_cell_line_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000001</td>\n",
       "      <td>NIHOVCAR3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000002</td>\n",
       "      <td>HL60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000003</td>\n",
       "      <td>CACO2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000004</td>\n",
       "      <td>HEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000005</td>\n",
       "      <td>HEL9217</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    DepMap_ID stripped_cell_line_name\n",
       "0  ACH-000001               NIHOVCAR3\n",
       "1  ACH-000002                    HL60\n",
       "2  ACH-000003                   CACO2\n",
       "3  ACH-000004                     HEL\n",
       "4  ACH-000005                 HEL9217"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_info = pd.read_csv(sample_info_file)\n",
    "# sample_info.set_index(['DepMap_ID'], inplace=True)\n",
    "sample_info = sample_info[['DepMap_ID', 'stripped_cell_line_name']]\n",
    "\n",
    "print(sample_info.shape)\n",
    "sample_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "e6d84ef5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NAs in the DepMap_ID indexes:  Counter({False: 1825})\n",
      "NAs in the columns:  Counter({False: 2})\n",
      "NAs in DataFrame:  1\n",
      "Row(s) with NA: \n",
      "        DepMap_ID stripped_cell_line_name\n",
      "1674  ACH-002260                     NaN\n",
      "New # of NAs in DataFrame:  0\n",
      "Dim of the DataFrame:  (1824, 2)\n"
     ]
    }
   ],
   "source": [
    "# Check for NAs in index\n",
    "print('NAs in the DepMap_ID indexes: ', collections.Counter(list(sample_info.index.isna())))\n",
    "\n",
    "# Check for NAs in columns\n",
    "print('NAs in the columns: ', collections.Counter(list(sample_info.columns.isna())))\n",
    "\n",
    "# Check for NaN in DataFrame\n",
    "print('NAs in DataFrame: ', sample_info.isnull().sum().sum())\n",
    "\n",
    "# Subset rows with NAs\n",
    "print('Row(s) with NA: \\n', sample_info[sample_info.isnull().any(axis=1)])\n",
    "\n",
    "# Drop NAs\n",
    "sample_info.dropna(subset = [\"stripped_cell_line_name\"], inplace=True)\n",
    "\n",
    "# Sanity Check\n",
    "print('New # of NAs in DataFrame: ', sample_info.isnull().sum().sum())\n",
    "print('Dim of the DataFrame: ',sample_info.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "3e5141ec",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1824, 3)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\fuzan\\Miniconda3\\envs\\compbio\\lib\\site-packages\\ipykernel_launcher.py:6: FutureWarning: The default value of regex will change from True to False in a future version.\n",
      "  \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DepMap_ID</th>\n",
       "      <th>stripped_cell_line_name</th>\n",
       "      <th>cleaned_stripped_cell_line_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACH-000001</td>\n",
       "      <td>NIHOVCAR3</td>\n",
       "      <td>NIHOVCAR3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACH-000002</td>\n",
       "      <td>HL60</td>\n",
       "      <td>HL60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACH-000003</td>\n",
       "      <td>CACO2</td>\n",
       "      <td>CACO2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACH-000004</td>\n",
       "      <td>HEL</td>\n",
       "      <td>HEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACH-000005</td>\n",
       "      <td>HEL9217</td>\n",
       "      <td>HEL9217</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    DepMap_ID stripped_cell_line_name cleaned_stripped_cell_line_name\n",
       "0  ACH-000001               NIHOVCAR3                       NIHOVCAR3\n",
       "1  ACH-000002                    HL60                            HL60\n",
       "2  ACH-000003                   CACO2                           CACO2\n",
       "3  ACH-000004                     HEL                             HEL\n",
       "4  ACH-000005                 HEL9217                         HEL9217"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_info[\"cleaned_stripped_cell_line_name\"] = sample_info[\"stripped_cell_line_name\"] \n",
    "\n",
    "# 1) force all names to be uppercase \n",
    "sample_info[\"cleaned_stripped_cell_line_name\"] = sample_info[\"cleaned_stripped_cell_line_name\"].str.upper()\n",
    "# 2) remove all special characters before comparing\n",
    "sample_info[\"cleaned_stripped_cell_line_name\"] = sample_info[\"cleaned_stripped_cell_line_name\"].str.replace('[^A-Za-z0-9]+','')\n",
    "# 3) remove NCI\n",
    "sample_info[\"cleaned_stripped_cell_line_name\"] = sample_info[\"cleaned_stripped_cell_line_name\"].str.replace('NCI','')\n",
    "\n",
    "print(sample_info.shape)\n",
    "sample_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "83369451",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1103, 5)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\fuzan\\Miniconda3\\envs\\compbio\\lib\\site-packages\\ipykernel_launcher.py:1: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>bio_context_id</th>\n",
       "      <th>cleaned_bio_context_id</th>\n",
       "      <th>DepMap_ID</th>\n",
       "      <th>stripped_cell_line_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1098</th>\n",
       "      <td>PH-FTX-IVP-2021-006-007_G8_27</td>\n",
       "      <td>22RV1</td>\n",
       "      <td>22RV1</td>\n",
       "      <td>ACH-000956</td>\n",
       "      <td>22RV1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1099</th>\n",
       "      <td>PH-FTX-IVP-2021-006-007_G8_29</td>\n",
       "      <td>LNCaP</td>\n",
       "      <td>LNCAP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1100</th>\n",
       "      <td>PH-FTX-IVP-2021-006-007_G8_56</td>\n",
       "      <td>LNCaP</td>\n",
       "      <td>LNCAP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1101</th>\n",
       "      <td>PH-FTX-IVP-2021-006-007_G8_7</td>\n",
       "      <td>22RV1</td>\n",
       "      <td>22RV1</td>\n",
       "      <td>ACH-000956</td>\n",
       "      <td>22RV1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1102</th>\n",
       "      <td>PH-FTX-IVP-2021-006-007_G8_9</td>\n",
       "      <td>LNCaP</td>\n",
       "      <td>LNCAP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             sample bio_context_id cleaned_bio_context_id  \\\n",
       "1098  PH-FTX-IVP-2021-006-007_G8_27          22RV1                  22RV1   \n",
       "1099  PH-FTX-IVP-2021-006-007_G8_29          LNCaP                  LNCAP   \n",
       "1100  PH-FTX-IVP-2021-006-007_G8_56          LNCaP                  LNCAP   \n",
       "1101   PH-FTX-IVP-2021-006-007_G8_7          22RV1                  22RV1   \n",
       "1102   PH-FTX-IVP-2021-006-007_G8_9          LNCaP                  LNCAP   \n",
       "\n",
       "       DepMap_ID stripped_cell_line_name  \n",
       "1098  ACH-000956                   22RV1  \n",
       "1099         NaN                     NaN  \n",
       "1100         NaN                     NaN  \n",
       "1101  ACH-000956                   22RV1  \n",
       "1102         NaN                     NaN  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_df = metadata.merge(sample_info, how='left', left_on='cleaned_bio_context_id', right_on='cleaned_stripped_cell_line_name').drop('cleaned_stripped_cell_line_name',1)\n",
    "\n",
    "print(sample_df.shape)\n",
    "sample_df.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "70c7a984",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>bio_context_id</th>\n",
       "      <th>cleaned_bio_context_id</th>\n",
       "      <th>DepMap_ID</th>\n",
       "      <th>stripped_cell_line_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>H202SC20082729_G12M41</td>\n",
       "      <td>MV-4-11</td>\n",
       "      <td>MV411</td>\n",
       "      <td>ACH-000045</td>\n",
       "      <td>MV411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>H202SC20082729_G12M42</td>\n",
       "      <td>MV-4-11</td>\n",
       "      <td>MV411</td>\n",
       "      <td>ACH-000045</td>\n",
       "      <td>MV411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>H202SC20082729_G12M43</td>\n",
       "      <td>MV-4-11</td>\n",
       "      <td>MV411</td>\n",
       "      <td>ACH-000045</td>\n",
       "      <td>MV411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>H202SC20082729_G12M44</td>\n",
       "      <td>MV-4-11</td>\n",
       "      <td>MV411</td>\n",
       "      <td>ACH-000045</td>\n",
       "      <td>MV411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>H202SC20082729_G14M49</td>\n",
       "      <td>MV-4-11</td>\n",
       "      <td>MV411</td>\n",
       "      <td>ACH-000045</td>\n",
       "      <td>MV411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>PH-FTX-IVP-2021-006-007_G7_38</td>\n",
       "      <td>22RV1</td>\n",
       "      <td>22RV1</td>\n",
       "      <td>ACH-000956</td>\n",
       "      <td>22RV1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>PH-FTX-IVP-2021-006-007_G7_51</td>\n",
       "      <td>22RV1</td>\n",
       "      <td>22RV1</td>\n",
       "      <td>ACH-000956</td>\n",
       "      <td>22RV1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>PH-FTX-IVP-2021-006-007_G8_18</td>\n",
       "      <td>22RV1</td>\n",
       "      <td>22RV1</td>\n",
       "      <td>ACH-000956</td>\n",
       "      <td>22RV1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1098</th>\n",
       "      <td>PH-FTX-IVP-2021-006-007_G8_27</td>\n",
       "      <td>22RV1</td>\n",
       "      <td>22RV1</td>\n",
       "      <td>ACH-000956</td>\n",
       "      <td>22RV1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1101</th>\n",
       "      <td>PH-FTX-IVP-2021-006-007_G8_7</td>\n",
       "      <td>22RV1</td>\n",
       "      <td>22RV1</td>\n",
       "      <td>ACH-000956</td>\n",
       "      <td>22RV1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>120 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             sample bio_context_id cleaned_bio_context_id  \\\n",
       "59            H202SC20082729_G12M41        MV-4-11                  MV411   \n",
       "60            H202SC20082729_G12M42        MV-4-11                  MV411   \n",
       "61            H202SC20082729_G12M43        MV-4-11                  MV411   \n",
       "62            H202SC20082729_G12M44        MV-4-11                  MV411   \n",
       "63            H202SC20082729_G14M49        MV-4-11                  MV411   \n",
       "...                             ...            ...                    ...   \n",
       "1093  PH-FTX-IVP-2021-006-007_G7_38          22RV1                  22RV1   \n",
       "1096  PH-FTX-IVP-2021-006-007_G7_51          22RV1                  22RV1   \n",
       "1097  PH-FTX-IVP-2021-006-007_G8_18          22RV1                  22RV1   \n",
       "1098  PH-FTX-IVP-2021-006-007_G8_27          22RV1                  22RV1   \n",
       "1101   PH-FTX-IVP-2021-006-007_G8_7          22RV1                  22RV1   \n",
       "\n",
       "       DepMap_ID stripped_cell_line_name  \n",
       "59    ACH-000045                   MV411  \n",
       "60    ACH-000045                   MV411  \n",
       "61    ACH-000045                   MV411  \n",
       "62    ACH-000045                   MV411  \n",
       "63    ACH-000045                   MV411  \n",
       "...          ...                     ...  \n",
       "1093  ACH-000956                   22RV1  \n",
       "1096  ACH-000956                   22RV1  \n",
       "1097  ACH-000956                   22RV1  \n",
       "1098  ACH-000956                   22RV1  \n",
       "1101  ACH-000956                   22RV1  \n",
       "\n",
       "[120 rows x 5 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_df.loc[sample_df[\"stripped_cell_line_name\"].isin(['22RV1', 'MV411'])]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f85db7bc",
   "metadata": {},
   "source": [
    "#### Save metadata to csv table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "c93e284b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save table to csv\n",
    "sample_df.to_csv('input_data/sample_df.csv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
